sibyl/learn/generator/pythongenerator.py

```diff
@@ -0,0 +1,297 @@
+            n=number))
+            addPass = False
+
+        if refsInMem:
+            self.printer.add_block(refUpdateTemplate.format(refs=repr(refsInMem), n=number))
+            addPass = False
+
+        if argList:
+            self.printer.add_block(argInitTemplate.format(argList=", ".join(argListStr(arg) for arg in argList), n=number))
+            addPass = False
+
+        if addPass:
+            self.printer.add_block("pass")
+
+        self.printer.sub_lvl()
+
+
+    def generate_check(self, snapshot, number):
+        '''Return the string corresponding to the code of the check function'''
+
+        refsInMem = {}
+        refs = snapshot.refs
+        if refs:
+            for addr, ref in refs.iteritems():
+                if ref.out_mem:
+                    refsInMem[addr] = ref.out_mem
+
+        # Remove arguments from out memory
+        memO = snapshot.out_memory
+        for i in xrange(self.nb_arg):
+            if i >= len(self.reg_list):
+                (stackOff, stackSeg) = snapshot.getStackSegment()
+
+                argAddr = stackOff + self.ptr_size * (i - len(self.reg_list)) + self.ptr_size
+
+                for (offset, seg) in memO.keys():
+                    addr = (offset, seg)
+                    if seg != stackSeg:
+                        continue
+
+                    if offset <= argAddr < offset + memO[addr].size:
+                        # Delete argument from memory not to check it manually
+                        if argAddr + self.ptr_size < offset + memO[addr].size:
+                            memaccess = MemoryAccess(memO[addr].size - (argAddr + self.ptr_size - offset),
+                                                     memO[addr].data[argAddr - offset:argAddr + self.ptr_size - offset],
+                                                     memO[addr].access,
+                                                     stackSeg)
+                            memO[(argAddr + self.ptr_size, stackSeg)] = memaccess
+
+                        if argAddr > addr:
+                            memO[addr].size = argAddr - addr
+
+                        else:
+                            del memO[addr]
+
+
+        addrList = []
+        for addr, mem in snapshot.out_memory.iteritems():
+            addrList.append((addr, mem.data))
+
+        res = snapshot.isRegInOutputRef(self.ira.ret_reg.name)
+        if res is None:
+            res = snapshot.output_reg[self.ira.ret_reg.name]
+
+        self.printer.add_block(funcDefTemplate.format("check", number))
+
+        self.printer.add_lvl()
+
+        self.printer.add_block(checkResultTemplate.format(expectedResult=res, n=number))
+
+        if addrList:
+            if refsInMem:
+                self.printer.add_block(checkAllocWithRefTemplate.format(addrList=addrList, n=number,refs=repr(refsInMem)))
+            else:
+                self.printer.add_block(checkAllocWithoutRefTemplate.format(addrList=addrList, n=number))
```

Comment:
*Cosmetic*: the generator outs sometime decimal, sometime hexadecimal number.
It could be great to uniform the output.
For instance, on the `my_strlen` test, in `init`:
```Python
allocList = [((0x5c4, 0...
```
and in `check`:
```Python
addrList = [((1476, 0...
```
It is just preferable to have the same way to write address when debug or modification are needed.